### PR TITLE
New attack technique: SES enumeration activities

### PR DESCRIPTION
**Pull request by loresuso**
### What does this PR do?

New attack technique

### Motivation

Attackers may use AWS SES to send phishing emails from a victim's AWS account. One of the objectives of the attackers could be sending as many emails as possible from a verified source to start a phishing campaign. Before doing that, and as described by[ this research ](https://www.invictus-ir.com/news/the-curious-case-of-dangerdev-protonmail-me), they can call some APIs for enumeration, like `ses:GetSendQuota` and `ses:ListIdentities`

### Checklist

- [x] The attack technique emulates a single attack step, not a full attack chain
- [x] We have factual evidence & references that the attack technique was used by real malware, pentesters, or attackers
- [x] The attack technique makes no assumption about the state of the environment prior to warming it up

### CODE REVIEW

**Comment by christophetd** [comment]
Thanks for the PR! Will review it in the coming days

**Comment by loresuso** [comment]
Thank you @christophetd!

File: v2/internal/attacktechniques/aws/discovery/ses-enumerate/main.tf [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  }
+}
+
+locals {
+  resource_prefix = "stratus-red-team-ses-enumerate"
+}
+
+data "aws_caller_identity" "current" {}
+
+resource "aws_iam_role" "ses_enumerate_role" {
```
Comment (christophetd) [inline v2/internal/attacktechniques/aws/discovery/ses-enumerate/main.tf:26]: Any specific reason why we need this role?

Comment (loresuso) [reply v2/internal/attacktechniques/aws/discovery/ses-enumerate/main.tf:26]: Similarly to the other discovery attack techniques, we just wanted a role to attach the `AmazonSESReadOnlyAccess` policy to

Comment (christophetd) [reply v2/internal/attacktechniques/aws/discovery/ses-enumerate/main.tf]: Which one(s) specifically? I think some other attacks create a role specifically to get access denies errors

**Comment by christophetd** [comment]
When an attacker wants to enumerate SES, they can achieve the same outcome in two different ways:

1. Use sesv2:GetAccount and read quotas in place and whether e-mail sending is enabled from the response
2. Use ses:GetAccountSendingEnabled + ses:GetSendQuota individually

Not sure which one we should be using, any thoughts? I've also asked in the Cloud Security Forum Slack

**Comment by christophetd** [comment]
@loresuso I've made a few changes to make enumeration slightly more aggressive, are you comfortable with the changes and can you have a look? Thanks!

**Comment by loresuso** [comment]
We are aware of the two alternatives, but initially thought to use GetSendQuota just because the GetAccount API call wasn't mentioned in the article we linked. Thinking about it, GetAccount should be better from an attacker's point of view because:
- it will produce fewer logs (just one call instead of two)
- it can provide even more additional details in the response, so it is better for enumeration

**Comment by loresuso** [comment]
By the way, thanks, the changes look good to me, especially the additional step with the `GetIdentityVerificationAttributes` call

**Comment by loresuso** [comment]
I think everything is good, when you want we can get this merged :) @christophetd 

I also have another question: I believe we could also implement another technique where we try to send an email using `ses:SendEmail` API. What do you think? How would you categorize it in MITRE ATT&CK? Curious to know your opinion about it!

**Comment by christophetd** [comment]
If we can find evidence that it's been abused in the wild (e.g. write-up) yes it would be perfect

**Pull request merged**